File: programmers/LV1/최소직사각형.js
// 최소직사각형
function solution(sizes) {
  const change = sizes.map(([w, h]) => (w < h ? [h, w] : [w, h]));
  let max = [0, 0];

  change.forEach(([w, h]) => {
    if (w > max[0]) max[0] = w;
    if (h > max[1]) max[1] = h;
  });

  return max[0] * max[1];
}

File: programmers/python/최소직사각형.py
# 최소직사각형
def solution(sizes):
    sizes = [sorted(s) for s in sizes]
    return max([x[0] for x in sizes]) * max([x[1] for x in sizes])





def solution2(sizes):
    return max(max(x) for x in sizes) * max(min(x) for x in sizes)